### DISCUSSION TITLE
[@types/video.js] Unable to get current version to work

### DISCUSSION THREAD

**Discussion by ghost** [DefinitelyTyped/DefinitelyTyped#65641] [Issues with a @types package] [answered]
There seems to have been updates recently (which is good), but now I can't get @types/video.js to work with a current Video.js 8.3.0 anymore.

I'm using the following:
```
import type { VideoJsPlayer } from "@types/video.js";
```

Which returns the following Typescript error:
```
Type 'Player' is missing the following properties from type 'VideoJsPlayer': bigPlayButton, controlBar, errorDisplay, liveTracker, and 18 more.
```

Am I holding it wrong? Any chance we could get a simple **How To** on npmjs.com?

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
> There seems to have been updates recently (which is good), but now I can't get @types/video.js to work with a current Video.js 8.3.0 anymore.
> 
> I'm using the following:
> 
> ```
> import type { VideoJsPlayer } from "@types/video.js";
> ```
> 
> Which returns the following Typescript error:
> 
> ```
> Type 'Player' is missing the following properties from type 'VideoJsPlayer': bigPlayButton, controlBar, errorDisplay, liveTracker, and 18 more.
> ```
> 
> Am I holding it wrong? Any chance we could get a simple **How To** on npmjs.com?

What's the version of your Video.js? If 8.*, please do not use @types/video.js.
From version 8.* vide.js ships it's own types:
https://github.com/videojs/video.js/releases/tag/v8.0.0
https://github.com/videojs/video.js/pull/7954

So just remove @types/video.js from reps.

Thanks!

**Reply by ghost** [reply]
I'm using 8.3.0 (see above)

Are you sure the current version adds any of the missing types? Things like `player.textTracks()` or `player.on("canplay", ...)` are in the docs and do work, but still cause TS errors. Nothing seems to have changed.

The badish, unsearchable documentation ain't exactly helping things either. I might redo the entire thing from scratch one of these days and see if that helps.

**Reply by peterblazejewicz** [reply]
I'm not sure, if I follow. The video.js ships typescript declaration files on it's starting version 8. 
Older declaration files are left on DT for older version. When using version 8.* of video.js, just do not install @types/video.js in your setup, thanks!

**Reply by ghost** [reply]
Like I said, I am using 8.3.0 and several methods cause TS errors. 8.3.0 either does not include types for everything or I am using it wrong - in which case a **How To** would be helpful.

**Reply by peterblazejewicz** [reply]
THis is community driven project, and only v7 are here. The 8.* module ships own types declaration, are those type of problem submitted to video.js?
And do not import anything directly from @types/video.js, those are only used to provide types declarations, so:
```ts
import *** from "video.js";
```
looks more like proper import for v8.*

**Reply by ghost** [reply]
Are you saying @types/video.js only works with VIDEOJS 7.x? Because that would make this useless for pretty much anyone. Could someone maybe put that info on Github or npmjs.com?

**Reply by peterblazejewicz** [reply, marked as answer]
I'm really sorry for your experience, but I think the goal of DT project is not to provide best experience on how to use video.js library itself. 
The owners of the Video.js are the people to address a problem of using v8 with TypeScript and that should be somehow addressed by the owners/people using Video.js, like on the intro page:
https://github.com/videojs/video.js#quick-start
Sorry, if you are getting confused how to use some open source project. DT repo contains only definition for version 7 (v7) of declaration files, the intent was to improve experience of people using VideoJS in TypeScript projects.
Thanks for understanding,

**Reply by ghost** [reply]
Okay, that's a pity. Thanks for the info!
